Add JDK14 to pipelines

AdamBrousseau opened the pull request:
- Remove refs to jdk12
- Reference repos
- Docker slaves
- PBT-All
- Variable file

[skip ci]
Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

AdamBrousseau commented:
@jdekonin for review.
Marking as draft until jdk14 repo is created.

pshipton commented:
> until jdk14 repo is created.

Scheduled for Dec 12th.

jdekonin commented on buildenv/jenkins/variables/defaults.yml:
```diff
@@ -157,203 +159,209 @@ s390x_linux:
+    14:
+      ? special.system
+    next:
+      ? special.system
```
We don't have the machine availability to handle these extra tests.  At least not until we have more AIX brought online.

jdekonin replied:
Or maybe this is irrelevant as we don't run these test mightly/daily.

AdamBrousseau replied:
We don't generally run `next` testing iirc. I was just adding the exclude to be more consistent.

AdamBrousseau commented:
Depends #8098 

Jenkins compile plinux jdk8,jdk11

pshipton commented on buildenv/jenkins/variables/defaults.yml:
```diff
@@ -538,46 +329,36 @@ x86-64_windows:
+    14:
+      - special.system
+    next:
+      - special.syste
```
missing an 'm' on system

AdamBrousseau replied:
fixed

AdamBrousseau commented:
Jenkins compile plinux jdk8,jdk11

AdamBrousseau commented:
Was using the 8/11 build to prove the change to `jdk_image_dir`
Doesn't work because `set_jdk_folder()` doesn't use buildspec manager. Will move this out to a different PR.

AdamBrousseau commented:
Jenkins test sanity.functional plinux jdk14

pshipton commented:
The compile failures (like `error: cannot access VarForm) are something that needs to be fixed for jdk14 independent of this change.

pshipton commented:
Jenkins test sanity.functional plinux jdk14

pshipton commented:
Something is wrong, there is a directory name missing
`/Build_JDK14_ppc64le_linux_Personal@tmp/durable-12659866/script.sh: build/linux-ppc64le-server-release/images//bin/java: not found`

AdamBrousseau commented:
I think I missed it because I originally had the changes from #8135 in this PR.
Fixed now. Also removed the excludes for 13+.
Jenkins test sanity.functional plinux jdk14

pshipton commented on buildenv/jenkins/variables/defaults.yml:
```diff
@@ -106,60 +106,47 @@ largeheap:
+# JITServer
+#========================================#
+jitserver:
+  extra_configure_options: '--enable-jitserver=yes'
```
Pls remove the `=yes` as per #8196

AdamBrousseau replied:
Done via #8098

pshipton commented on buildenv/jenkins/variables/defaults.yml:
```diff
@@ -286,163 +200,71 @@ s390x_linux_xl:
-    12: '--enable-cuda --with-cuda=/usr/local/cuda-9.0 --with-openssl=fetched'
-    13: '--enable-cuda --with-cuda=/usr/local/cuda-9.0 --with-openssl=fetched'
-    next: '--enable-cuda --with-cuda=/usr/local/cuda-9.0 --with-openssl=fetched'
+    all: '--enable-cuda --with-cuda=/usr/local/cuda-9.0'
```
This should include `--enable-jitserver`, assuming #8191 is merged first.

AdamBrousseau replied:
Based on https://github.com/eclipse/openj9/pull/8191#issuecomment-571298582 I will leave as-is for now. Please advise if you'd like it included or if 8191 is merged first.

pshipton commented on buildenv/jenkins/variables/defaults.yml:
```diff
@@ -600,59 +356,39 @@ x86-32_windows:
   excluded_tests:
     8:
       - special.system
-    11:
```
This shouldn't be adding Windows jdk11 special.system testing.

AdamBrousseau replied:
This is win32 so it doesn't apply which is why I am removing the exclude.

pshipton commented:
Note the copyright check failures

AdamBrousseau commented:
Updated copyrights and rebased again on master

pshipton commented:
Jenkins compile plinux jdk14
